src/reference_systems.jl

```diff
@@ -3,136 +3,216 @@ module ReferenceSystems
 
 using LinearAlgebra
 
+using StaticArrays: SVector, SMatrix
+
 using SOFA
 using SatelliteDynamics.Constants
 using SatelliteDynamics.Universe: UT1_UTC, POLE_LOCATOR
 using SatelliteDynamics.Time: Epoch, mjd
 
+const idx123    = SVector(1, 2, 3)
+const idx456    = SVector(4, 5, 6)
+const idx123456 = SVector(1, 2, 3, 4, 5, 6)
```

Comment:
These really should be placed at the top module level in `SatelliteDynamics.jl` and used consistently throughout all  submodule as to reduce reuse. But given that the StaticArray updates are confined to just this location, I'm fine leaving it here for now.

This is pedantic, but I would suggest renaming the variables `idx123` -> `idx1t3`, `idx456` -> `idx4t6` and `idx123456` -> `idx1t6` to make the variables consistent length and clearly ranges.

I would prefer not to have to use these const-defs at all, but I appreciate that they're necessary to preserve type stability and the performance improvements provided by StaticArrays, so I think the performance benefits in this case outweigh the readability concerns.

Reply:
> This is pedantic, but I would suggest renaming the variables `idx123` -> `idx1t3`, `idx456` -> `idx4t6` and `idx123456` -> `idx1t6` to make the variables consistent length and clearly ranges.

I agree, I pushed a new revision with this change, if you want.